core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -335,10 +336,26 @@ private[chisel3] class DynamicContext(
   val throwOnFirstError: Boolean) {
   val importDefinitionAnnos = annotationSeq.collect { case a: ImportDefinitionAnnotation[_] => a }
 
-  // Ensure there are no repeated names for imported Definitions
-  val importDefinitionNames = importDefinitionAnnos.map { a => a.definition.proto.name }
-  if (importDefinitionNames.distinct.length < importDefinitionNames.length) {
-    val duplicates = importDefinitionNames.diff(importDefinitionNames.distinct).mkString(", ")
+  // Map holding the actual names of extModules
+  val importDefinitionMap: mutable.Map[String, String] = mutable.Map.empty
```

Comment:
Can you make this immutable, built from iterating over importDefinitionAnnos, rather than a mutable map?

Reply:
Done.